cli.md

```diff
@@ -48,6 +49,7 @@ The `executable` could involve multiple command line tokens (e.g., `npm run` or
         - **“line”:** the codemod to which the parameter applies (optional — assumed “all” if not present)
         - **“name”:** the of the parameter (required)
         - **“value”:** the value of the parameter (required)
+- The `--max-threads` argument specifies the maximum number of threads to use for parallel codemod processing. Codemodders that support parallel processing should use a default that is related to the number of available cores (either `<num cores>` or `<num cores> - 1`). Codemodders that do not support parallel processing should use a default of `1` and ignore any other value that is provided.
```

Comment:
I think the default is not correct here since we can benefit from parallelism even on a single core, but I'm a little unclear on how else to specify a sensible default.

Reply:
Maybe `2 * <num cores>` to start? Or maybe we should just leave the default unspecified.

Reply:
Interesting from the Python docs:
> Changed in version 3.5: If max_workers is None or not given, it will default to the number of processors on the machine, multiplied by 5, assuming that [ThreadPoolExecutor](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor) is often used to overlap I/O instead of CPU work and the number of workers should be higher than the number of workers for [ProcessPoolExecutor](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ProcessPoolExecutor).

Reply:
On one hand, I'm not too concerned with the default, because I think Pixee platform will always specify this configuration, so I could go for "it's unspecified and the platform picks some reasonable default based on how parallelism works in that language".

Or I'm fine with some big-ish number that reflects our IO-bound-ness like 100.

Reply:
I don't think we can leave a default unspecified because it would require CLI users to pick a number. I'd be okay with saying "if not specified, codemodder is free to pick a reasonable default based on the available thread pool implementation". And/or say "An integer multiple of the number of cores is recommended"